src/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js

```diff
@@ -33,6 +36,67 @@ export default class geFormWidgetTokenizeCard extends LightningElement {
     CUSTOM_LABELS = GeLabelService.CUSTOM_LABELS;
     PAYMENT_TRANSACTION_STATUS_ENUM;
 
+    _currentPaymentMethod = undefined;
+    _hasPaymentMethodInTemplate = false;
+
+    @api
+    get widgetDataFromState() {
+        return this._widgetDataFromState;
+    }
+
+    set widgetDataFromState(widgetState) {
+        this._widgetDataFromState = widgetState;
+        this.handleWidgetDataChange(widgetState);
+    }
+
+    handleWidgetDataChange(widgetState) {
+        this._hasPaymentMethodInTemplate =
+            this.sourceFieldsUsedInTemplate.includes(apiNameFor(DATA_IMPORT_PAYMENT_METHOD));
+
+        if (this._hasPaymentMethodInTemplate) {
+            this._currentPaymentMethod = widgetState[apiNameFor(DATA_IMPORT_PAYMENT_METHOD)];
+
+            if (this.hasValidPaymentMethod(this._currentPaymentMethod)) {
+                if (this.isMounted) {
+                    this.requestSetPaymentMethod(this._currentPaymentMethod);
+                } else {
+                    this.handleUserEnabledWidget();
+                }
+            } else {
+                this.handleUserDisabledWidget();
+            }
+        }
+    }
+
+    hasValidPaymentMethod(paymentMethod) {
+        return paymentMethod === 'ACH' || paymentMethod === 'Credit Card';
```

Comment:
Could we add constants for 'ACH' and 'Credit Card'?

Reply:
Updated!